File: src/main/java/org/railwaystations/rsapi/core/model/InboxStateQuery.java
package org.railwaystations.rsapi.core.model;

import lombok.Builder;
import lombok.Value;

import java.time.Instant;

@Value
@Builder
public class InboxStateQuery {

    Long id;

    String countryCode;

    String stationId;

    String title;

    Coordinates coordinates;

    @Builder.Default
    InboxState state = InboxState.UNKNOWN;

    String comment;

    ProblemReportType problemReportType;

    String rejectedReason;

    String filename;

    String inboxUrl;

    Long crc32;

    Instant createdAt;

    public enum InboxState {
        UNKNOWN,
        REVIEW,
        CONFLICT,
        ACCEPTED,
        REJECTED
    }

}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/InboxStateQuery.kt
package org.railwaystations.rsapi.core.model

import java.time.Instant

class InboxStateQuery(
    var id: Long,
    var countryCode: String? = null,
    var stationId: String? = null,
    var title: String? = null,
    var coordinates: Coordinates? = null,
    var newTitle: String? = null,
    var newCoordinates: Coordinates? = null,
    var state: InboxState = InboxState.UNKNOWN,
    var comment: String? = null,
    var problemReportType: ProblemReportType? = null,
    var rejectedReason: String? = null,
    var filename: String? = null,
    var inboxUrl: String? = null,
    var crc32: Long? = null,
    var createdAt: Instant? = null,
) {

    enum class InboxState {
        UNKNOWN,
        REVIEW,
        CONFLICT,
        ACCEPTED,
        REJECTED
    }
}
